OptionSelector/CQTableViewController.m

```diff
@@ -0,0 +1,153 @@
+//
+//  CQTableViewController.m
+//  Unit1WeekFourHomework
+//
+//  Created by Jackie Meggesto on 8/12/15.
+//  Copyright (c) 2015 Jackie Meggesto. All rights reserved.
+//
+
+#import "CQTableViewController.h"
+#import "CQCategory.h"
+#import "CQDetailTableViewController.h"
+
+
+
+@interface CQTableViewController ()
+@property (nonatomic) NSMutableArray *categories;
+
+@end
+
+@implementation CQTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.categories = [[NSMutableArray alloc]init];
+    CQCategory *saurischia = [[CQCategory alloc]init];
+    CQCategory *ornithischia = [[CQCategory alloc]init];
+    CQCategory *militaryPlanes = [[CQCategory alloc]init];
+    saurischia.name = @"Saurischian Dinosaurs";
+    ornithischia.name = @"Ornithischian Dinosaurs";
+    militaryPlanes.name = @"Military Aircraft";
+    saurischia.options = @[
+                           @"Brachiosaurus",
+                           @"Apatosaurus",
+                           @"Camarasaurus",
+                           @"Argentinosaurus",
+                           @"Tarbosaurus",
+                           @"Tyrannosaurus",
+                           @"Crylophosaurus",
+                           @"Deinonychus"
+                           ];
+    ornithischia.options = @[
+                             @"Lambeosaurus",
+                             @"Corythosaurus",
+                             @"Styracosaurus",
+                             @"Chasmosaurus",
+                             @"Iguandodon",
+                             @"Psittacosaurus",
+                             @"Ankylosaurus",
+                             @"Nodosaurus"
+                             ];
+    militaryPlanes.options = @[
+                               @"AV-8B Harrier II",
+                               @"A-10 Thunderbolt II",
+                               @"F-15 Eagle",
+                               @"F-22 Raptor",
+                               @"F-35 Lightning II",
+                               @"Sukoi Su-37",
+                               @"MiG-29",
+                               @"F/A-18E Super Hornet"
+                               ];
+    [self.categories addObject:saurischia];
+    [self.categories addObject:ornithischia];
+    [self.categories addObject:militaryPlanes];
+    
+   
+}
+-(void)viewWillAppear:(BOOL)animated {
+    [self.tableView reloadData];
```

Comment:
Two comments here. First, when you're overriding methods like `viewDidLoad` or `viewWillAppear:`, you should always call the `super` implementation first:
```objc
[super viewWillAppear:animated];
//...
```
That way, you'll run the code in the superclass's implementation of this method, and then your code.

Second, you probably noticed that after you select an option in the detail view and then go back, there's a delay before any changes are reflected in the master table view. If you wanted the change to be more instantaneous, you could create a `delegate` property on your detail view controller and set it to `self` when preparing to segue from the master to the detail controller, and send a message to the master controller whenever an option was selected indicating that it should reload its table view.